sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1364,8 +1364,13 @@ static void _test_options(gnrc_netif_t *netif)
             assert(netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR);
             assert(netif->l2addr_len >= 3U && netif->l2addr_len <= 5U);
             break;
-        case NETDEV_TYPE_LORA: /* LoRa doesn't provide L2 ADDR */
         case NETDEV_TYPE_SLIP:
+#if IS_USED(MODULE_SLIPDEV_L2ADDR)
+            assert(netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR);
+            assert(8U == netif->l2addr_len);
+            break;
+#endif /* IS_USED(MODULE_SLIPDEV_L2ADDR) */
+        case NETDEV_TYPE_LORA: /* LoRa doesn't provide L2 ADDR */
```

Comment:
I think this introduced a bug. Shouldn't it have been:
```c
#if IS_USED(MODULE_SLIPDEV_L2ADDR)
            assert(netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR);
            assert(8U == netif->l2addr_len);
#endif /* IS_USED(MODULE_SLIPDEV_L2ADDR) */
            break;
        case NETDEV_TYPE_LORA: /* LoRa doesn't provide L2 ADDR */
```

(excluding the break from the guard)

Reply:
Before it was

```C
        case NETDEV_TYPE_LORA: /* LoRa doesn't provide L2 ADDR */
￼       case NETDEV_TYPE_SLIP:
            assert(!(netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR));
            assert(0U == netif->l2addr_len);
            /* don't check MTU here for now since I'm not sure the current
             * one is correct ^^" "*/
            break;
```

so the functionality stays the same without `slipdev_l2addr`

Reply:
ah I'm an idiot. I miss read the assert()s to be identical between the two cases - which they are not. Sorry!